numalogic/registry/mlflow_registry.py

```diff
@@ -249,6 +257,16 @@ def transition_stage(self, model_name: str) -> Optional[ModelVersion]:
                     version=str(version - 2),
                     stage=ModelStage.ARCHIVE,
                 )
+
+            # only keep "models_to_retain" number of models.
+            list_model_versions = list(
+                self.client.search_model_versions("name='{}'".format(model_name))
+            )
+            while len(list_model_versions) >= models_to_retain:
+                if len(list_model_versions) == models_to_retain:
+                    break
+                model_version_delete = list_model_versions.pop(0).version
+                self.delete(skeys=skeys, dkeys=dkeys, version=model_version_delete)
```

Comment:
let's use list slicing to select the versions to keep and delete. Will make things clearer

Reply:
good idea